specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-03-01/managedClusters.json

```diff
@@ -65,6 +65,49 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ContainerService/locations/{location}/os-options": {
+      "get": {
+        "tags": [
+          "ContainerServices"
+        ],
+        "operationId": "ContainerServices_GetOSOptions",
+        "summary": "Gets supported OS options in the specified subscription.",
+        "description": "Gets supported OS options in the specified subscription.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceTypeParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          },
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/OSOptionProfile"
```

Comment:
This URI is a GET collection API (listing all resource of the locations/osOptions resource type).  It must follow the GET collection schema.  If you only have one you'll only have one entry in the "value" array  (Paging response body here:  https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#get-resource)

Reply:
For the API, it will always return only one result object. Let me know if things need to be modified here.

Reply:
ARM will expect this to return a collection response `{ "value": [ {yourObjectGoesHere} ] }` due to its URI structure.  It is getting a collection of resources of type locations/osOptions.

Reply:
I don't see that you are following the appropriate format, this is a LIST call, you are also missing nextLink.

See an example [here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/deploymentScripts.json#L259) and the [definition](https://github.com/Azure/azure-rest-api-specs/blob/8830e8dc294fa2db59fa75081540d22b3fbfad34/specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/deploymentScripts.json#L812) is an array type.

In your case you are returning one object with an array property which is not what your api implies as Chris already pointed out. If you have only one result, still please adhere to the contract.

---
In reply to: [603611939](https://github.com/Azure/azure-rest-api-specs/pull/13619#discussion_r603611939) [](ancestors = 603611939)

Reply:
If this is a singleton, then add /default as the resource name.

---
In reply to: [603683051](https://github.com/Azure/azure-rest-api-specs/pull/13619#discussion_r603683051) [](ancestors = 603683051,603611939)

Reply:
Yes, the resource is a singleton, add /default in the path as commented.

Reply:
Great, now, to be fully compliant we expect RPs to expose a GET for collection and for a single resource, otherwise it might have an impact with Policy, see [here](https://armwiki.azurewebsites.net/api_contracts/guidelines/policy.html#plcy008-support-a-collection-get-and-individual-get-for-child-resource-types-even-singletons). You have two options:

1. Implement a GET for a collection (where you'll return 1 element always)
2. Add an ack from your manager (as a comment here) stating that you are ok with not providing a GET for a collection

---
In reply to: [603686873](https://github.com/Azure/azure-rest-api-specs/pull/13619#discussion_r603686873) [](ancestors = 603686873)